engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -181,15 +180,38 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     val actionThreadPoolSize = databaseConfig.getIntOr("actionThreadPoolSize", dbNumThreads) min dbMaximumPoolSize
     Executors.newFixedThreadPool(actionThreadPoolSize)
   }
+
   private val actionExecutionContext: ExecutionContext = ExecutionContext.fromExecutor(
-    actionThreadPool, database.executor.executionContext.reportFailure)
+    actionThreadPool, database.executor.executionContext.reportFailure
+  )
+
+  private lazy val useSlickUpserts = dataAccess.driver.capabilities.contains(JdbcProfile.capabilities.insertOrUpdate)
 
   override def close(): Unit = {
     actionThreadPool.shutdown()
     database.close()
   }
 
-  // Run action with an outer transaction
+  private def runTransactionWithRollbackRetry[R](action: DBIOAction[R, _ <: NoStream, _ <: Effect], actorSystem: ActorSystem): Future[R] = {
+    def retry[T](f: => Future[T], delay: FiniteDuration, retries: Int): Future[T] = {
+      implicit val ec = actorSystem.dispatcher
+      f recoverWith {
+        case e: MySQLTransactionRollbackException if retries > 0 =>
```

Comment:
I think I had it that way at some point, then I had some reason for changing it back to `MySQLTransactionRollbackException`... though I can't think of the reason for it